Custom NpmRegistry: Sequence contains no elements

# Bug Report

## Prerequisites

- [x] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [x] Are you running the latest version?
- [x] Did you perform a search in the issues?

## Environment Details and Version

**net8.0**
**Piral.Blazor.x** 8.0.11

## Description
Unable to use private npm registry for installing the Piral instance.

## Steps to Reproduce

1. `$ dotnet new blazorpilet --piralInstance my-app-shell --npmRegistry https://xxx`
2. Adding .npmrc file (following Azure artifacts guide - works on Angular/React Pilets when installing Piral instance)
3. `$ dotnet run`

## Expected behavior

For Pilet to run and being accessible in browser.

## Actual behavior

Error: `.nuget/packages/piral.blazor.tools/8.0.11/build/Piral.Blazor.Tools.targets(26,9): error : Sequence contains no elements`

A bit more context about the issue
1. Set `<PiralInstance>` to an existing instance that is published to a private npm registry (e.g. ADO)
2. Set `<NpmRegistry>` with the link to a private npm registry
3. Run build process

Console output
```
    [piral-cli] ≡ƒªï  piral-cli v1.4.3
1>        [piral-cli] - Preparing source and target ...
1>        [piral-cli] - Scaffolding new pilet in D:\Projects\blazorpilet\src\piral~\blazorpilet ...
1>        [piral-cli] - Setting up npm registry (https://private-registry/npm/registry/) ...
1>        [piral-cli] - Installing npm package my-app-shell@0.0.43 ...
1>        [piral-cli] Γ£û [0002] Could not install the package "my-app-shell@0.0.43" using npm. Make sure npm is correctly installed and accessible: Error: npm ERR! code E401
1>        [piral-cli] npm ERR! Unable to authenticate, your authentication token seems to be invalid.
1>        [piral-cli] npm ERR! To correct this please try logging in again with:
1>        [piral-cli] npm ERR!     npm login
1>        [piral-cli]
1>        [piral-cli] npm ERR! A complete log of this run can be found in: C:\Users\user\AppData\Local\npm-cache\_logs\2024-02-12T13_32_05_296Z-debug-0.log
1>        [piral-cli] npm ERR! code E401
1>        [piral-cli] npm ERR! Unable to authenticate, your authentication token seems to be invalid.
1>        [piral-cli] npm ERR! To correct this please try logging in again with:
1>        [piral-cli] npm ERR!     npm login
1>        [piral-cli]
1>        [piral-cli] npm ERR! A complete log of this run can be found in: C:\Users\user\AppData\Local\npm-cache\_logs\2024-02-12T13_32_05_296Z-debug-0.log
1>        [piral-cli] ≡ƒÉ¢  Find detailed descriptions at https://docs.piral.io/code/search
1>        [piral-cli]
1>        [piral-cli]
1>        C:\Users\user\.nuget\packages\piral.blazor.tools\7.0.12\build\Piral.Blazor.Tools.targets(26,9): warning : The provided command 'cmd.exe /c npx.cmd pilet new "my-app-shell@0.0.43" --registry "https://private-registry/npm/registry/" --bundler none --no-install' in 'D:\Projects\blazorpilet\..\piral~\blazorpilet' did not finish successfully. The process might be corrupted.
1>        Successfully updated 'package.json'.
```

The issue is caused by calling "npx pilet new ***' before copying `.npmrc`

--- a/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
+++ b/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
@@ -201,13 +201,13 @@ namespace Piral.Blazor.Tools
 
             if (File.Exists(existingConfigFile))
             {
-                Log.LogWarning($"Ignoring NpmRegistry property! Copying file '{configFileName}' from '{existingConfigFile}'.");
+                Log.LogMessage($"Taking '{configFileName}' from '{existingConfigFile}'.");
                 var targetFile = Path.Combine(target, configFileName);
                 File.Copy(existingConfigFile, targetFile, true);
             }
             else
             {
-                Log.LogMessage($"Creating '{configFileName}' with provided NpmRegistry property.");
+                Log.LogMessage($"Creating '{configFileName}' with entry from NpmRegistry.");
                 File.AppendAllLines(Path.Combine(target, configFileName), [$"registry={NpmRegistry}", "always-auth=true"], Encoding.UTF8);
             }
         }

--- a/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
+++ b/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
@@ -191,24 +191,24 @@ namespace Piral.Blazor.Tools
             return 0;
         }
 
-        private void CopyConfigurationFiles()
+        private void InitializeNpmConfiguration()
         {
-            var configurationFiles = new[] { ".npmrc" };
+            var configFileName = ".npmrc";
             var source = ConfigDir;
             var target = ProjectDir;
 
-            Log.LogMessage($"Copying config files from '{source}'...");
+            var existingConfigFile = Path.Combine(source, configFileName);
 
-            foreach (var configurationFile in configurationFiles)
+            if (File.Exists(existingConfigFile))
             {
-                var sourceFile = Path.Combine(source, configurationFile);
-
-                if (File.Exists(sourceFile))
-                {
-                    var targetFile = Path.Combine(target, configurationFile);
-                    Log.LogMessage($"Copying file '{configurationFile}' from '{sourceFile}'.");
-                    File.Copy(sourceFile, targetFile, true);
-                }
+                Log.LogWarning($"Ignoring NpmRegistry property! Copying file '{configFileName}' from '{existingConfigFile}'.");
+                var targetFile = Path.Combine(target, configFileName);
+                File.Copy(existingConfigFile, targetFile, true);
+            }
+            else
+            {
+                Log.LogMessage($"Creating '{configFileName}' with provided NpmRegistry property.");
+                File.AppendAllLines(Path.Combine(target, configFileName), [$"registry={NpmRegistry}", "always-auth=true"], Encoding.UTF8);
             }
         }
 
@@ -403,10 +403,13 @@ namespace Piral.Blazor.Tools
             }
 
             Directory.CreateDirectory(target);
+
+            InitializeNpmConfiguration();
+
             Run("npm", npm, target, $"{npmPrefix}init -y");
             Run("npm", npm, target, $"{npmPrefix}install piral-cli@{CliVersion} --save-dev");
             Run("npm", npm, target, $"{npmPrefix}install piral-cli-{Bundler}@{bundlerVersion} --save-dev");
-            Run("piral-cli", npx, target, $"{npxPrefix}pilet new \"{Emulator}\" --registry \"{NpmRegistry}\" --bundler none --no-install");
+            Run("piral-cli", npx, target, $"{npxPrefix}pilet new \"{Emulator}\" --bundler none --no-install");
 
             ChangeOutputDirectory();
 
@@ -456,7 +459,7 @@ namespace Piral.Blazor.Tools
             }
 
             var json = JObject.Parse(File.ReadAllText(packageJsonFile));
-            
+
             if (!string.IsNullOrEmpty(Version))
             {
                 json["version"] = Version;
@@ -510,7 +513,7 @@ namespace Piral.Blazor.Tools
         {
             if (string.IsNullOrEmpty(name))
             {
-                name = "pilet";                
+                name = "pilet";
             }
 
             name = name.ToLowerInvariant();
@@ -564,7 +567,7 @@ namespace Piral.Blazor.Tools
                 throw new Exception($"The file '{originalJsonFile}' does not exist.");
             }
 
-            if (!File.Exists(overwritesJsonFile)) 
+            if (!File.Exists(overwritesJsonFile))
             {
                 Log.LogMessage($"No '{target}' file found to merge into '{source}'.");
                 return;
@@ -576,10 +579,10 @@ namespace Piral.Blazor.Tools
             }
 
             var result = new JObject();
-            var originalJson = JObject.Parse(File.ReadAllText(backup)); 
-            var overwritesJson = JObject.Parse(File.ReadAllText(overwritesJsonFile)); 
+            var originalJson = JObject.Parse(File.ReadAllText(backup));
+            var overwritesJson = JObject.Parse(File.ReadAllText(overwritesJsonFile));
 
-            result.Merge(originalJson); 
+            result.Merge(originalJson);
             result.Merge(overwritesJson);
 
             File.WriteAllText(originalJsonFile, JsonConvert.SerializeObject(result, Formatting.Indented));
@@ -617,8 +620,6 @@ namespace Piral.Blazor.Tools
 
                 if (canScaffold)
                 {
-                    CopyConfigurationFiles();
-                    
                     var requireInstall = PreparePilet();
 
                     if (requireInstall)

--- a/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
+++ b/src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs
@@ -617,9 +617,9 @@ namespace Piral.Blazor.Tools
 
                 if (canScaffold)
                 {
-                    var requireInstall = PreparePilet();
-
                     CopyConfigurationFiles();
+                    
+                    var requireInstall = PreparePilet();
 
                     if (requireInstall)
                     {

